src/SFA.DAS.Employer.PR.Web/Controllers/Requests/RequestsController.cs

```diff
@@ -63,15 +81,32 @@ public async Task<IActionResult> PostRequestDetails([FromRoute] Guid requestId,
         {
             GetPermissionRequestResponse permissionRequest = await _outerApiClient.GetPermissionRequest(requestId, cancellationToken);
 
-            EmployerAccountCreationModel viewModel = GetViewModel(permissionRequest, submitModel.HasAcceptedTerms);
+            var changeNameLink = Url.RouteUrl(RouteNames.CreateAccountChangeName, new { requestId });
+
+            var sessionModel = _sessionService.Get<AccountCreationSessionModel>();
+            SetNamesInSessionModel(sessionModel, permissionRequest);
+
+            EmployerAccountCreationViewModel viewModel = GetViewModel(permissionRequest, changeNameLink!);
             result.AddToModelState(ModelState);
             return View(RequestsCheckDetailsViewPath, viewModel);
         }
 
+        _sessionService.Delete<AccountCreationSessionModel>();
+
         return RedirectToRoute(RouteNames.CreateAccountCheckDetails, new { requestId, submitModel.HasAcceptedTerms });
     }
 
 
+
+    private static void SetNamesInSessionModel(AccountCreationSessionModel? sessionModel, GetPermissionRequestResponse? permissionRequest)
```

Comment:
GetPermissionRequestResponse should be not nullable here

Reply:
ok